### PR TITLE
Enhance parser with typing, test suite, configurable debug and pyproject.toml fix

### PR DESCRIPTION
**Pull request by JaskRendix**
PR introduces several improvements:
- adds full type annotations across the parser module for better clarity and tooling support  
- introduces a comprehensive unit test suite to validate parsing behavior and edge cases  
- refactors the `parse()` method to allow configurable `debug` and `write_tables` options  
- fixes an issue in `pyproject.toml` where the README was incorrectly set to `Readme.rst`; it now correctly points to `README.md`
- adds a command-line interface (CLI) for parsing IDF files directly via terminal  
- introduces a `parse_file()` helper for streamlined file-based parsing with error handling  
- enhances error reporting with clearer diagnostics for syntax, file and unexpected issues  
- prints object type counts and total parsed objects for quick validation after parsing

### CODE REVIEW

File: parseidf.py [hunk 1]
```diff
@@ -22,104 +23,178 @@
...
+    :param write_tables: Passed to the parser's write_tables setting.
+    :return: A dictionary mapping object types (str, uppercased) to lists
+        of their field value lists.
+    """
+    path = Path(file_path)
+
+    try:
+        idf_text = path.read_text(encoding="utf-8")
+    except FileNotFoundError:
+        print(f"Error: File not found at {file_path}")
```
Comment (daren-thomas) [inline parseidf.py]: this kind of printing should be in the `__main__` block - that way, the module can be used as a library in other tools.

**Review by daren-thomas — CHANGES_REQUESTED** [review]

File: parseidf.py (hunk 1 above)
Comment (JaskRendix) [reply parseidf.py]: @daren-thomas Good point, I moved all the printing and user‑facing messages into the `__main__` block and left the library functions to just raise exceptions. That way the module stays clean to import and only the CLI handles output, as you suggested.
